rust/candid/src/parser/value.rs

```diff
@@ -301,15 +301,15 @@ impl IDLValue {
             IDLValue::Float32(_) => Type::Float32,
             IDLValue::Float64(_) => Type::Float64,
             IDLValue::Text(_) => Type::Text,
-            IDLValue::None => Type::Opt(Box::new(Type::Null)),
+            IDLValue::None => Type::Opt(Box::new(Type::Empty)),
```

Comment:
The line below is reasonable, but this line looks fishy.

Also, would it be worth extending the test suite with a case to catch this, to help other implementations? Or does the reference test suite not cover this code path?

Reply:
Why is this a problem? A `null : opt t` can use any type `t`.

The reference test suite doesn't cover this code path. This is only for serializing untyped textual values.

Reply:
Ah, ok. It looked as if it would lead to `null : empty` which would be wrong.

Reply:
Hmm, no, it doesn't even look like that, I was just to hasy while reading. All good, ignore me :-)